awx/main/models/events.py

```diff
@@ -602,7 +602,7 @@ def _update_host_summary_from_stats(self, hostnames):
                     h.last_job_host_summary_id = host_mapping[h.id]
                     updated_hosts.add(h)
 
-            Host.objects.bulk_update(list(updated_hosts), ['last_job_id', 'last_job_host_summary_id'], batch_size=100)
+            Host.objects.bulk_update(sorted(updated_hosts, key=lambda host: host.id), ['last_job_id', 'last_job_host_summary_id'], batch_size=100)
```

Comment:
For longer-term maintainability, it might be nice to introduce a utility method. Even if the method seems trivial (a simple `sorted` action), it would help stitch these two pieces of code together, and establish the expectation for the future.

And actually, I might even use that here:

https://github.com/ansible/awx/blob/db6e8b9bad660bdc028f48e8461e6794b6c3cef1/awx/main/tasks/host_metrics.py#L150

That might not _currently_ risk contention with other queries, but it is a bulk_update with a batch_size. Seems like there may be a good argument that _all_ such queries in the app should use consistent ordering, because it completely fixes the Deadlock issue.